feat(dark-mode): add dark mode feature

Kishan1835 opened the pull request:
Dark Mode feature added to the whole Page 
![Screenshot 2025-03-22 221257](https://github.com/user-attachments/assets/1c2c192a-05a9-4c07-aee4-b4dfb22ab068)

![Screenshot 2025-03-22 221450](https://github.com/user-attachments/assets/6d66d5f8-0e45-4962-bd05-8ec0eafe05f2)
![Screenshot 2025-03-22 221501](https://github.com/user-attachments/assets/60b1925c-031d-4a8f-afa6-79b52604cc3b)

JustinBenito commented:
Wow, this looks stellar @Kishan1835 🔥 
Love how the design has come to darkmode. 
There are few suggestions from my side reg this. 

One would be to change the github contribute button on top to text-white in dark mode. 
the other would be to make the calltoaction.tsx file updated with respect to the new PR pulled by @MadhavanAR

JustinBenito commented on src/components/EmptyEventCard.tsx:
```diff
@@ -0,0 +1,17 @@
+import { CalendarX } from '@phosphor-icons/react';
+
```
This seems to be a copy of the no-events-card.tsx which already exists, can I please know why you introduced the same here @Kishan1835 ?

also was curious to know why an empty ClientThemeProvider has been added ?

JustinBenito commented on src/components/Tooltip.tsx:
```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+
+interface TooltipProps {
```
Also can I know why we are abstracting just this tooltips component into a separate file, I understand it helps readability but apart from that is there any specific requirement in doing this @Kishan1835  ?

JustinBenito commented on tailwind.config.ts:
```diff
@@ -1,17 +1,19 @@
+  darkMode: 'class', // Enable dark mode using the 'class' strategy
+  content: [
+    './src/pages/**/*.{js,ts,jsx,tsx,mdx}',
+    './src/components/**/*.{js,ts,jsx,tsx,mdx}',
```
also just curious if there is a specific idea behind adding only few directories ?

@Kishan1835

JustinBenito reviewed (changes requested):
Overall the new PR looks promising, but there are few gaps which would be really great if we could fill it. 
Can you please look into the given suggestions @Kishan1835 and update with a new PR.

Cheers :)

P.S also can you please include a video showing all the new changes, so that it would be great to understand the new changes. 

Thanks

Kishan1835 commented:
@JustinBenito Thank you for the feedback! I’ll definitely look into the suggestions and make the necessary updates soon. Appreciate your time reviewing the PR!

Sarveshmrao commented:
Hey @Kishan1835 any update on this PR?

Kishan1835 commented:
@JustinBenito @Sarveshmrao 
I have updated the PR as per the feedback:
• Confirmed text-white for the Contribute button.
• Renamed EmptyEventcard to NoEventsCard and updated references.
• Retained the Tooltip abstraction for reusability.
• Verified that the Tailwind config covers all src files.
Awaiting details on MadhavanAR's PR for calltoaction.tsx .
Kindly review at your convenience.
